torch/csrc/jit/fusion_compiler.cpp

```diff
@@ -416,6 +416,11 @@ void CompiledFusionFunction::launch(uint32_t numel, void ** arguments) {
   //std::cout << "maxBlocks = " << maxBlocks << " needed blocks: " << ceilDiv(numel,blockSize)
   //          << " numblocks =  " << numBlocks;
 
+  // it is possible that this is the first cuda call on this thread
+  // so make sure we initialize the Driver API's context
+  // cudaFree(0) accomplishes this.
+  cudaFree(0);
```

Comment:
We have to check that this is not a GPU sync point before this is merged

Reply:
According to https://stackoverflow.com/questions/12539593/is-cudafree-asynchronous cudaFree is synchronous.

Reply:
While cudaFree can sometimes block, cudaFree(0) doesn't do anything but initialize the context and (unfortunately) has become the idiomatic way to do this: https://devtalk.nvidia.com/default/topic/911817/cudamalloc-is-too-expensive-and-gpu-memories/?offset=6